<?xml version="1.0"?>

<!-- rover_gazebo_keyboard_ctrl.launch

Use Gazebo to simulate and control a rover with Ackermann steering
using keyboard commands.

Arguments:
  namespace (string, default: rover)
    rover namespace.
-->

<launch>
  <arg name="namespace" default="rover"/>
  <arg name="cmd_timeout" default="0.5"/>
  <arg name="node_start_delay" default="5"/>
  <arg name="publish_frequency" default="30.0"/>
  <arg name="max_speed" default="10.0"/>
  <arg name="max_angle" default="1.0"/>
  <arg name="topic_name" default="ackermann_cmd"/>

  <!-- Launch the Gazebo simulation of the rover. -->
  <include file="$(find rover_gazebo_control)/launch/rover_gazebo_control.launch">
    <arg name="namespace" value="$(arg namespace)"/>
  </include>

  <!-- Control the steering and wheel joints. -->
  <node name="rover_controller" pkg="rover_gazebo_control" type="rover_controller.py"
        output="screen" ns="$(arg namespace)">
    <param name="cmd_timeout" value="$(arg cmd_timeout)"/>
    <param name="publish_frequency" value="$(arg publish_frequency)"/>
    <rosparam file="$(find rover_gazebo_control)/config/rover_ackermann_ctrlr_params.yaml" command="load"/>
  </node>

  <node name="rover_keyop" pkg="rover_gazebo_control" type="rover_keyop.py"
        output="screen" ns="$(arg namespace)" args="$(arg max_speed) $(arg max_angle) $(arg topic_name)"/>
</launch>
